build-hypre/action.yml

```diff
@@ -30,25 +30,43 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Get Hypre (Windows)
+      if: runner.os == 'Windows'
+      run: |
+        $URI_HYPRE = "${{ inputs.url }}/${{ inputs.archive }}"
+        Invoke-WebRequest -Uri $URI_HYPRE -OutFile ${{ inputs.archive }}
+        Remove-Item ${{ inputs.dir }} -Force -Recurse -ErrorAction Ignore;
+        tar -xzf ${{ inputs.archive }};
+      shell: pwsh
+
+    - name: Get Hypre (Linux/macOS)
+      if: runner.os != 'Windows'
+      run: |
+        wget --no-verbose ${{ inputs.url }}/${{ inputs.archive }};
+        rm -rf ${{ inputs.dir }};
+        tar -xzf ${{ inputs.archive }};
+      shell: bash
+
     - name: Install Hypre
       run: |
         echo "Map target to options"
         if [[ "${{ inputs.target }}" == "int32" ]]
         then
-          export hypre_options="--disable-fortran";
+          export hypre_options="";
         elif [[ "${{ inputs.target }}" == "int64" ]]
         then
-          export hypre_options="--disable-fortran --enable-bigint";
+          export hypre_options="-DHYPRE_ENABLE_BIGINT=ON";
         else
           echo "Hypre target not recognized";
           exit 1;
         fi
         echo "Installing Hypre"
-        wget --no-verbose ${{ inputs.url }}/${{ inputs.archive }};
-        rm -rf ${{ inputs.dir }};
-        tar -xzf ${{ inputs.archive }};
-        cd ${{ inputs.dir }}/src;
-        ./configure ${hypre_options} CC=mpicc CXX=mpic++;
-        make -j3;
```

Comment:
I know it will be a pain, but I think I'd prefer to retain the makefile build of hypre on Mac and Linux.

(their CMake option may not be as well supported)

Reply:
I think in the interim, we can revert this and just use vcpkg to install hypre for Windows builds. Long-term, we might want to consider using Spack to build these dependencies.

Reply:
Sounds good

Reply:
Turns out building mfem with the vcpkg version of hypre causes the unit tests to fail. (maybe something to do with vcpkg's version of lapack?)

I've added back in the CMake build for Windows only; for Mac and Linux, we'll just continue to build with the makefile system.

Reply:
ok, that's what I was thinking originally

Reply:
I designed these actions so that they introduce as little "smart choices" as possible, keeping them as generic as possible. I wanted to have an "interface" with parameters to control the behavior of the install at will.

For that reason, I would have created a new parameter "build-system" so that MFEM CI can control which build system it wants for hypre explicitely, instead of hiding it in the hypre install action with a "smart choice" depending on the OS.

Again, the reason for that is to keep those action generic, so that they can be used in other projects (mfem related, radiuss related, or not), and avoid duplication.

Would you agree with that vision?